skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -0,0 +1,596 @@
+import numpy as np
+from skimage import util
+cimport numpy as np
+cimport cython
+from libc.math cimport exp
+
+ctypedef np.float32_t DTYPE_t
```

Comment:
Well, this was an attempt to imitate the cython tutorial http://docs.cython.org/src/tutorial/numpy.html#adding-types 
I can indeed change the name of the alias.